Diary/Data/PersistentStorage/DiaryStorage/DiaryStorage.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  CoreDataStorage.swift
+//  Diary
+//
+//  Created by Jaehun Lee on 5/6/24.
+//
+
+import Foundation
+import CoreData
+
+final class DiaryStorage {
+    private let coreDataStorage: CoreDataStorage
+    
+    init(coreDataStorage: CoreDataStorage = CoreDataStorage.shared) {
+        self.coreDataStorage = coreDataStorage
+    }
+    
+    func loadDiaries() async throws -> [Diary] {
+        let taskContext = coreDataStorage.newTaskContext()
+        let request: NSFetchRequest<DiaryEntity> = DiaryEntity.fetchRequest()
+        
+        request.predicate = NSPredicate(format: "TRUEPREDICATE")
+        request.sortDescriptors = [NSSortDescriptor(key: "editedDate", ascending: false)]
+        
+        return try await taskContext.perform {
+            try taskContext.fetch(request).map { $0.toDomain() }
```

Comment:
요건 제 취향인데 ㅎㅎ 
toDomain이 함수가 아니라 computed property라면 keypath로도 쓸 수 있어요

```suggestion
            try taskContext.fetch(request).map(\.domain)
```